src/main/java/seedu/address/model/person/fields/Address.java

```diff
@@ -42,6 +42,13 @@ public static boolean isValidAddress(String test) {
         return test.matches(VALIDATION_REGEX);
     }
 
+    /**
+     * Returns true if test string is contained within the value of the field.
+     */
+    public boolean contains(String test) {
+        return this.value.toUpperCase().contains(test.toUpperCase());
+    }
```

Comment:
As mentioned in `Person.java`, this method can be abstracted into a `Field` abstract class or interface (same applies for the rest of the fields).

> although if that's too much work I can understand up to your own discretion there's so much work this week and so little time aaaaahhhhhh